src/ape/types/trace.py

```diff
@@ -225,3 +229,768 @@ class TraceFrame(BaseInterfaceModel):
     """
     The raw trace frame from the provider.
     """
+
+
+class SourceContent(BaseModel):
+    """
+    A wrapper around source code line numbers mapped to the content
+    string of those lines.
+    """
+
+    __root__: Dict[int, str]
+
+    @property
+    def begin_lineno(self) -> int:
+        return self.line_numbers[0] if self.line_numbers else -1
+
+    @property
+    def end_lineno(self) -> int:
+        return self.line_numbers[-1] if self.line_numbers else -1
+
+    @property
+    def line_numbers(self) -> List[int]:
+        """
+        All line number in order for this piece of content.
+        """
+        return sorted(list(self.__root__.keys()))
+
+    def items(self):
+        return self.__root__.items()
+
+    def as_list(self) -> List[str]:
+        return list(self.__root__.values())
+
+    def __getitem__(self, lineno: int) -> str:
+        return self.__root__[lineno]
+
+    def __iter__(self):
+        yield from self.__root__
```

Comment:
this would iter the line numbers?

Reply:
updated this to be cooler and still work